### PR TITLE
Fix sqlite3 queues

### PR DESCRIPTION
**Pull request by Digenis**
Work in progress for bug #150

### CODE REVIEW

**Comment by Digenis** [comment]
@dkachan1941, this should fix your issue correctly.
You can test this by installing #147 with pip:
`git+https://github.com/Digenis/scrapyd.git@backport_bugfixes_to_1.1`
Don't forget to uninstall scrapyd first.
If this works for you too,
it will be in the next 1.1 bugfix release soon.

**Comment by Digenis** [comment]
Any chance trying this, @dkachan1941

**Comment by dkachan1941** [comment]
Hi. 
I unninstalled old scrapyd and installed https://github.com/Digenis/scrapyd.git with pip.
I got:
2016-06-17 16:56:59+0300 [_GenericHTTPChannelProtocol,0,127.0.0.1] Unhandled Error
	Traceback (most recent call last):
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/twisted/web/http.py", line 1767, in allContentReceived
	    req.requestReceived(command, path, version)
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/twisted/web/http.py", line 768, in requestReceived
	    self.process()
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/twisted/web/server.py", line 183, in process
	    self.render(resrc)
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/twisted/web/server.py", line 234, in render
	    body = resrc.render(self)
	--- <exception caught here> ---
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/scrapyd/webservice.py", line 17, in render
	    return JsonResource.render(self, txrequest)
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/scrapyd/utils.py", line 19, in render
	    r = resource.Resource.render(self, txrequest)
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/twisted/web/resource.py", line 250, in render
	    return m(request)
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/scrapyd/webservice.py", line 44, in render_POST
	    spiders = get_spider_list(project, version=version)
	  File "/home/spider/scrapyV1/venv/lib/python2.7/site-packages/scrapyd/utils.py", line 116, in get_spider_list
	    raise RuntimeError(msg.splitlines()[-1])
	exceptions.RuntimeError: Use "scrapy" to see available commands

**Comment by Digenis** [comment]
The command for installing with pip is
`pip install git+https://github.com/Digenis/scrapyd.git@backport_bugfixes_to_1.1`,
I guess you used `pip install https://github.com/Digenis/scrapyd.git` instead
which default to the (wrong) master branch.

If you had the write version installed, line 44 in webservice.py should be
`def render_POST(self, txrequest):`
and not `spiders = get_spider_list(project, version=version)`.

**Comment by Digenis** [comment]
@dkachan1941, any chance of trying this before I go on with scrapyd 1.1.1?
You are the only one that can reproduce the bug.

File: scrapyd/sqlite.py [hunk 1]
```diff
@@ -60,26 +64,26 @@ def items(self):
...
 class JsonSqliteDict(SqliteDict):
 
     def encode(self, obj):
-        return json.dumps(obj)
+        return sqlite3.Binary(json.dumps(obj))
 
-    def decode(self, text):
-        return json.loads(text)
+    def decode(self, obj):
+        return json.loads(bytes(obj))
```
Comment (kmike) [inline scrapyd/sqlite.py]: What is bytes(obj) for? What type can obj be?

Comment (Digenis) [inline scrapyd/sqlite.py]: A buffer.

**Comment by Digenis** [comment]
I amended the last commit
because of a leftover import statement
that sneaked through `git add -p`.

If there are no more comments I'd like to merge this soon.

**Pull request merged**